### PR TITLE
feat: inlay hints displayStyle

### LINKED ISSUE #145191: an option for more compact inlay hints
rust-analyzer recently switched to using LSP inlay hints, which render using vscode's inlay hint renderer. with this involves a change in design,  and most notably, an introduction of some slight padding around the inlay hint. 

in rust-analyzer, inlay hints where it make sense try to look like the actual inferred type definition in many cases, for example:

```rust
let i = String::from("hello");
````

Might insert the following inlay hint:

```rust
let i: String = String::from("hello");
     ^^^^^^^^
```

In vscode today, however, there is some awkward padding around this:

image

In an older rust-analyzer build, this used to look like (taken from r-a's github page which still shows the old style.)

image

An effort can be made to try to reduce the visual impact of the inlay hints by setting `editorInlayHint.background` and `editorInlayHint.foreground` - however, this still does not solve the issue of the awkward padding box that is now hidden:

image

This suffers from odd padding on the left/right side, and the bottom. 

It would be really much appreciated if vscode could support a "compact" display style option which would remove the padding, border radius and vertical alignment.

The "compact" version might look like this (still probably a pixel or two off, crafted in mspaint..., but should be enough to show the idea)

image

I would be happy to implement this feature too if pointed in the right direction!

### PR DESCRIPTION
**Pull request by jhgg**
fixes #145191 

This PR implements "compact" inlay hint display style, which removes the padding/border radius/vertical alignment as discussed in #145191

### CODE REVIEW

File: src/vs/editor/common/config/editorOptions.ts [hunk 1]
```diff
@@ -2514,6 +2514,13 @@ export interface IEditorInlayHintsOptions {
 	 * Defaults to editor font family.
 	 */
 	fontFamily?: string;
+
+	/**
+	 * The display style to render inlay hints with.
+	 * Compact mode disables the borders and padding around the inlay hint.
+	 * Defalts to 'standard'.
```
Comment (jogo-) [inline src/vs/editor/common/config/editorOptions.ts:2521]: Typo: Defalts -> Defa**u**lts

File: src/vs/monaco.d.ts [hunk 2]
```diff
@@ -3724,6 +3724,12 @@ declare namespace monaco.editor {
 		 * Defaults to editor font family.
 		 */
 		fontFamily?: string;
+		/**
+		 * The display style to render inlay hints with.
+		 * Compact mode disables the borders and padding around the inlay hint.
+		 * Defalts to 'standard'.
+		 */
+		displayStyle: 'standard' | 'compact';
```
Comment (jogo-) [inline src/vs/monaco.d.ts:3732]: Typo: Defalts -> Defa**u**lts

File: src/vs/monaco.d.ts [hunk 3]
```diff
@@ -3724,6 +3724,12 @@ declare namespace monaco.editor {
 		 * Defaults to editor font family.
 		 */
 		fontFamily?: string;
+		/**
+		 * The display style to render inlay hints with.
+		 * Compact mode disables the borders and padding around the inlay hint.
+		 * Defalts to 'standard'.
```
Comment (jogo-) [inline src/vs/monaco.d.ts:3730]: Typo: Defalts -> Defa**u**lts

**Review by jrieken — APPROVED** [review]
lgtm

**Pull request merged**